src/lib/io/sol-iio.c

```diff
@@ -1107,10 +1111,12 @@ channel_get_offset(struct sol_iio_channel *channel)
 
 #define GET_OFFSET(_name) \
     do { \
-        r = craft_filename_path(path, sizeof(path), CHANNEL_OFFSET_PATH, \
-            device->device_id, _name); \
-        if (r && sol_util_read_file(path, "%d", &channel->offset) > 0) { \
-            goto end; \
+        if (_name) { \
```

Comment:
why are you doing this check here ? Shouldn't you handle properly when `channel->name` is `NULL` ? This check will possibly avoid a crash here but the next instruction is `channel_get_pure_name` that crashes when `channel->name` is `NULL`